Topic4/src/tests/ShoppingCartTest.java

```diff
@@ -0,0 +1,274 @@
+package tests;
+
+import static org.junit.Assert.*;
+
+import java.util.List;
+
+import interfaces.Item;
+import interfaces.PaymentStrategy;
+import interfaces.ShoppingCart;
+import interfaces.Transaction;
+import org.junit.Test;
+import classes.CashStrategy;
+import classes.CreditCardStrategy;
+import classes.FunctionalShoppingCart;
+import classes.Offer;
+import classes.PaypalStrategy;
+import classes.Product;
+import classes.WebTransaction;
+
+public class ShoppingCartTest {
+
+	@Test
+	public void createShoppingCart() {

```

Comment:
Even when it is redundant, try to add the word test in the name of each test (as prefix or postfix)